core/src/main/java/io/grpc/ClientCallImpl.java

```diff
@@ -85,27 +108,21 @@
     ClientTransport get();
   }
 
+  private enum State {INIT, STARTED, HALF_CLOSED, CANCELED}
+
   ClientCallImpl<ReqT, RespT> setUserAgent(String userAgent) {
     this.userAgent = userAgent;
     return this;
   }
 
   @Override
-  public void start(Listener<RespT> observer, Metadata.Headers headers) {
-    Preconditions.checkState(stream == null, "Already started");
+  public void start(ClientCall.Listener<RespT> observer, Metadata.Headers headers) {
```

Comment:
A general question to ask that generally leads to good results:

1. Is the change necessary for this PR?

Things that can conflict with that rule:

1. Is the change a non-trivial dependent change that would be most easily reviewed in a separate PR?

"Most easily reviewed" also includes things like "lower risk" such that you don't block one set of changes that everyone wants due to another that there is more contention about.

Style changes have higher risk and lower return, and they tend to simply be noise during review, which does hinder the time it takes to go through the review process. The changing of the state checking from booleans to enums would be a riskier thing to include, especially if it is completely orthogonal. Removing Preconditions would be similar high risk and low return.

However, limited variable renaming because you are adding a new field can improve readability and can reduce confusion. But that assumes that you changed something in the PR that made the problem worse than it was previously (like adding a new class field that has same/similar name), and that the added review noise won't make things worse for the reviewer than the rename itself helps. So that means doing a rename that touches 1000 lines of code and then having a functional change also in the PR is not going to work out well, since the rename churn makes it too hard to find the actual functional change.

Things are easier for new code than changing code. For new code the author gets to choose the approach and the reviewer should really defer to your judgement unless there are strong reasons against. For existing code, the author needs to have the reason, and that reason needs to be _stronger_ than the reasons to leave the code as-is, lest you get into constant code flip-flops and churn.